src/main/kotlin/org/opensearch/replication/action/stop/TransportStopIndexReplicationAction.kt

```diff
@@ -149,7 +150,12 @@ class TransportStopIndexReplicationAction @Inject constructor(transportService:
         }
     }
 
-    private fun validateStopReplicationRequest(request: StopIndexReplicationRequest) {
+    private fun validateStopReplicationRequest(state: ClusterState, request: StopIndexReplicationRequest) {
+        if (state.blocks.hasIndexBlock(request.indexName, INDEX_REPLICATION_BLOCK)
+                || state.metadata.index(request.indexName)?.settings?.get(REPLICATED_INDEX_SETTING.key) != null) {
+            return
+        }
+
```

Comment:
Can we have separate method to validate state? The name says request, but we are validating state.

Can we also leave some helpful comment as to when this can happen?

Reply:
Ack. Modified the method name (to reflect replication state).